src/components/list.js

```diff
@@ -1,47 +1,54 @@
-import React, { Fragment } from 'react';
+import React, { Component, Fragment } from 'react';
 import { connect } from 'react-redux';
 import { Redirect, Link } from 'react-router-dom';
+import { getItems, toggleChecked } from '../actions/items';
 
-function handleClick(event) {
-  return {};
-}
-const strikeThrough = { textDecoration: 'line-through' };
-export function List(props) {
-  if (props.authLoading) {
-    return <div>Logging In...</div>;
+export class List extends Component {
+  onClickHandler(itemId) {
+    this.props.dispatch(toggleChecked(itemId));
   }
-  if (!props.username) {
-    return <Redirect to="/" />;
+  componentDidMount() {
+    this.props.dispatch(getItems(this.props.listId));
   }
-  const items = props.list.items.map(item => {
+  render() {
+    if (this.props.authLoading || this.props.list.loading) {
+      return <div>Loading...</div>;
+    }
+    if (!this.props.username) {
+      return <Redirect to="/" />;
+    }
+    const items = this.props.list.items.map(item => {
+      return (
+        <li
+          key={item.id}
+          style={item.checked ? { textDecoration: 'line-through' } : null}
```

Comment:
Generally, I prefer to use classes/className to change the style of something, rather than using the style prop. Personal choice, but at some point we should come to a convention about it.

Reply:
noo totally. i was just being lazy. inline styles are bad design. I can update that

Reply:
nvm. i'm not sure how to do that